README
# Übersicht

## Beachten wenn man diesen Repository erstmals clonen will

- Zuerst muss alles via wsl home verzeichnis alles gezogen werden durch "git clone [URL]". 

## Checklist bevor man weiterarbeitet

- Bevor man generell weiterarbeitet, sollte immer der "git pull" befehl im Terminal ausgeführt werden (im Falle das ein User neue Änderungen vorgenommen hat)

- Dann schaut man ob die Repository Links im "Branch Verzeichnis" verknüpft ist (Für die Visuelle Veranschaulichung)

- Zum Anschluss kann man an den Daten weiterarbeiten in VSCode

## Checklist wenn man fertig ist für den Moment

- "Lokal" (Strg + S) Speichern in vscode

- Im Terminal mit "git add [Dateinamen]" die Datei für den Commit bereitstellen

- Bevor man commited, schaut man mit "git status" nach, ob die Dateien auch alle bereitgestellt sind für den Commit

- Sowie alles stimmt, verwendet man "git commit -m "kurzer changelog mit den vorgenommenen Änderungen""

- Anschließend schaut man entweder mit "git log" im Terminal nach wie der Verlauf der Commits ist oder grafisch mit "gitg" (sofern installiert)

# WICHTIG

- Alle steps müssen über die linux shell "WSL" ausgeführt werden!!!!

Changelog
# Changelog

## 02.11.2023 - Adams & Meyer
- Ordnerstruktur angefangen zu erstellen
- Raumbezeichnungen abgeändert anhand der Raumnamen im MapsMaker Programm
    - Physisches Abbild bei Adams am Platz
    - Speicherdatei der Map auf Nextcloud
- Notwendige Dateien erstellt bzw. bereitgestellt

test.c
#include <stdio.h>
#include <stdlib.h>
#include <stdbool.h>
#include "inventar.h"
#include "Story.c"
/*
Kürzel Siehe ReadMe Datei
*/

const char key3[30] = "keycard 3";

// SCP ASCII Funktion Definierung

void SCP(const char *dateiname)
{
    FILE *datei = fopen(dateiname, "r");

    if (datei)
    {
        char zeile[256];
        while (fgets(zeile, sizeof(zeile), datei))
        {
            printf("%s", zeile);
        }
    
        fclose(datei);
    }
    else
    {
        printf("\n\nDie Datei '%s' konnte nicht geöffnet werden, \n", dateiname);
    }
}

// Spielstart
void startGame() 
{
    printf("\n \nSpiel wird gestartet: \n \n");
    r1_s();
}

//Korrosives Loch

void easter_egg_larry()
{
    const char *dateiname2 = "Larry.txt";

    printf("\n\nDu springst elegant hinein und landest perfekt.\n");
    printf("Du schaust dich um und siehst SCP-106 (auch Larry genannt).\n\n");
    SCP(dateiname2);
    printf("\n\nLarry war nicht erfreut das du in sein Loch gesprungen bist und zögert nicht lange dich zu töten.\n\n");
    printf("Game Over (Du hast Larry gefunden! :D)\n\n");
    printf("Infos zu SCP-106 sind im folgenden Link → ");
    printf("file:///mnt/C/Users/adamsdav/Nextcloud/FIAE2023-1/Gruppenprojekte/SCP_Text-Based-Adventure_Game_Adams_Meyer_Huebscher/SCP-textadventure/SCP-Liste/SCP-106/SCP-106.html \n\n");
}

//Fehlermeldung sowie eine ungültige Eingabe getätigt wurde

void debug()
{
    const char *dateiname1 = "peanut.txt";
    SCP(dateiname1);
    printf("\n\nSCP-173 hat gesehen was du versucht hast und fand das nicht so toll...\n\n");
}

// Für Entscheidungssituationen

int path2_o (char choice)
{
    if (choice == 'j')
    {
        return true;
    }
    else
    if (choice == 'n')
    {
        return false;
    }
}

int path4_o (char choice)
{
    if(choice == 'l')
    {

    }
    else if (choice == 'v')
    {
        return true;
    }
    else if(choice == 'r')
    {

    }
    else if(choice == 'o')
    {
        return true;
    }
}

int path6_o (char choice)
{
    if(choice == 'v')
    {
        return true;
    }
    else if (choice == '1')
    {
        return true;
    }
    else if(choice == '2')
    {
        return true;
    }
    else if(choice == '3')
    {
        return true;
    }
    else if(choice == '4')
    {
        return true;
    }
    else if(choice == 'z')
    {
        return true;
    } 
}

//clear Funktion um das terminal aufzuräumen
void clear(){
    #if defined(__linux__) || defined(__unix__) || defined(__APPLE__)
        system("clear");
    #endif

    #if defined(_WIN32) || defined(_WIN64)
        system("cls");
    #endif
    }


int main()
{
    int choices;
    Main_Menu();
    scanf("%d",&choices);

switch (choices)
{
    case 1:    
        char* playerLocation[99];
        char choice;
        startGame();
        playerLocation[0] = "r1_s";
        scanf("%s",&choice);

    for(int i = 0; i != 99; i++)
    {
        //Raum h1_s
        if(strcmp(*playerLocation, "r1_s") == 0)
        {
            if(path2_o(choice))
            {
                h1_s();
                scanf("%s", &choice);
                playerLocation[0] = "h1_s";
            }
            else
            {
                r1_d_2();
                exit(1);
            }
        }
        
        //Raum h1_d_1
        if(strcmp(*playerLocation, "h1_s") == 0)
        {
            if(path2_o(choice))
            {
                h1_d_1();
                scanf("%s", &choice);
                playerLocation[0] = "h1_d_1";
            }
            else
            {
                h1_d_2();
                exit(2);
            }
        }

        //Raum h1_d_1 decision nach welchem Raum
        if(strcmp(*playerLocation, "h1_d_1") == 0)
        {
            if(path4_o(choice))
            {
                if(choice == 'l')
                {
                    scanf("%s", &choice);
                }
                else if(choice == 'v')
                {
                    addItem(key3, 1);
                    removeItem(key3);
                    h1_1_d_s();
                    scanf("%s", &choice);
                    playerLocation[0] = "h1_1_d_s";
                }
                else if(choice == 'r')
                {
                    scanf("%s", &choice);
                }
                else if(choice == 'o')
                {
                    easter_egg_larry();
                    exit(3);
                }
            }
        }

        //Raum h1_1_d_s decision zu welchem Raum
        if (strcmp(*playerLocation, "h1_1_d_s") == 0)
        {
            if(path6_o(choice))
            {
                if(choice == 'v')
                {
                    scanf("%s", &choice);                    
                }
                else if (choice == 'L' && strcmp(inventory[0].name, "keycard 3") == true)
                {
                    printf("lol\n");
                    scanf("%s", &choice);
                }
                else if(choice == 'l')
                {
                    scanf("%s", &choice);
                }
                else if(choice == '3')
                {
                    scanf("%s", &choice);
                }
                else if(choice == '4')
                {
                    scanf("%s", &choice);
                }
                else if(choice == 'z')
                {
                    h1_d_1();
                    scanf("%s", &choice);
                    playerLocation[0] = "h1_d_1";
                } 
            }
        }

    }





    
    break;

    case 2:
    credits();
    break;

    case 3:
    printf("Das Programm wird nun beendet \n");
    exit(0);
    break;

    default:
    debug();
    break;
return 0;
}
}

src/test/test.c
#include <stdio.h>
#include <stdlib.h>
#include <stdbool.h>
#include "../../lib/gameMechanics.c"
/*
Kürzel Siehe ReadMe Datei
*/

const char key3[] = "keycard 3";


int main()
{
    int choices;
    Main_Menu();
    scanf("%d",&choices);

switch (choices)
{
    case 1:    
        char* playerLocation[99];
        char choice;
       // int choices;

        do
        {
        if(choices == 1)
        {
            choices = 0;
        }

            startGame();
            playerLocation[0] = "r1_s";
            scanf("%s",&choice);

        for(int i = 0; i != 99; i++)
        {
            //Raum h1_s
            if(strcmp(*playerLocation, "r1_s") == 0)
            {
                if(path2_o(choice))
                {
                    room2_3();
                    scanf("%s", &choice);
                    playerLocation[0] = "h1_s";
                }
                else
                {
                    starting_room_d_2();
                    exit(1);
                }
            }
            
            //Raum h1_d_1
            if(strcmp(*playerLocation, "h1_s") == 0)
            {
                if(path2_o(choice))
                {
                    room4();
                    scanf("%s", &choice);
                    playerLocation[0] = "h1_d_1";
                }
                else
                {
                    room2_3_d_2();
                    exit(2);
                }
            }

            //Raum h1_d_1 decision nach welchem Raum
            if(strcmp(*playerLocation, "h1_d_1") == 0)
            {
                if(path4_o(choice))
                {
                    if(choice == 'l')
                    {
                        scanf("%s", &choice);
                    }
                    else if(choice == '2')
                    {
                        room2scps();
                        scanf("%s", &choice);
                        playerLocation[0] = "h1_1_d_s";
                    }
                    else if(choice == '3')
                    {
                        scanf("%s", &choice);
                    }
                    else if(choice == '4' || choice == 'o')
                    {
                        easter_egg_larry();
                        exit(3);
                    }
                }
            }

            //Raum room2scps decision zu welchem Raum
            if (strcmp(*playerLocation, "room2scps") == 0)
            {
                if(path6_o(choice))
                {
                    if(choice == 'v')
                    {
                        scanf("%s", &choice);                    
                    }
                    else if (choice == 'L' && strcmp(inventory[0].name, "keycard 3") == 0)
                    {
                        printf("lol\n");
                        scanf("%s", &choice);
                    }
                    else if(choice == 'l')
                    {
                        scanf("%s", &choice);
                    }
                    else if(choice == '3')
                    {
                        scanf("%s", &choice);
                    }
                    else if(choice == '4')
                    {
                        scanf("%s", &choice);
                    }
                    else if(choice == 'z')
                    {
                        room4();
                        scanf("%s", &choice);
                        playerLocation[0] = "h1_d_1";
                    } 
                }
            }
        }

        printf("Möchtest du es erneut versuchen? (1 = Ja)\n");
        scanf("%d", &choices);
    }
    while (choices == 1);

    break;

    case 2:
    credits();
    break;

    case 3:
    printf("Das Programm wird nun beendet \n");
    exit(0);
    break;

    default:
    debug();
    break;
return 0;
}
}

lib/Story.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

/*Für die Namensgebung siehe playerLocation.h*/

void starting_room()
{
    printf("\nDu erwachst in einem Raum, der für dich wie eine Zelle aussieht. \n");
    printf("Wenn du dich umschaust, erkennst du die Leichen von deinen einstigen Kollegen, \n");
    printf("welche durch einen Genickbruch gestorben sind. \n");
    printf("Ohne dich näher zu fragen, was passiert ist, weißt du schon,\n");
    printf("dass du aus dieser Einrichtung entkommen musst. \n \n");
    printf("Vor dir siehst du eine Tür: Möchtest du diese öffnen und durchgehen? \n\n");

    printf("1. Ja \n2. Nein\n");
}

void starting_room_d_2()
{
    printf("\nDu bist in die Zelle zurück gegangen und wartest, was dazu geführt hat, \n");
    printf("dass du durch das MTF (Mobile Task Force) getötet wurdest.\n");
    printf("\n\nGame Over.\n\n");
}

void room2_3()
{
    printf("\nDu hast dich entschieden, die Tür zu öffnen und siehst einen leeren Flur. \n");
    printf("PLÖTZLICH bebt der ganze Komplex und du hörst durch einen Lautsprecher eine Stimme, \n");
    printf("die sagt:\"Achtung an das gesamte Personal! Meldet euch unverzüglich bei [Störsignal]\". \n");
    printf("Voraus siehst du eine weitere verschlossene Tür, die du öffnen könntest.\n\n");
    printf("Traust du dich diese Tür zu öffnen?\n");

    printf("1. Ja \n2. Nein \n");
}

void room2_3_d_2()
{
printf("\nDu hast dich dagegen entschieden, was dazu geführt hat, dass eine Kachel, die sich durch das Beben gelockert hat, dir auf den Kopf gefallen ist.\n");
printf("Die Kachel hat sich durch die Wucht in deinen Schädel gebohrt und bist dadurch gestorben.\n");
printf("Wärst du nicht so feige gewesen und durch die Tür gegangen, wärst du wahrscheinlich noch am Leben...\n\n");

printf("Peinliches Game Over...(nicht einmal ein würdiges Ende...)\n\n");
}

void room4()
{
    printf("\nVorsichtig öffnest du die Tür und schaust dich um.\n");
    printf("Du siehst einen weiteren Flur mit einem großen Loch der nur vor sich her korridiert.\n");
    printf("Du fragst dich wie tief das Loch ist, bist dennoch nicht neugierig genug es herauszufinden.\n");
    printf("Beim weiteren umsehen kannst du 3 Türen erkennen, weißt aber nicht ob sich dahinter was verbirgt.\n");
    printf("\nAn welche der drei Türen möchtest du herantreten?\n\n");
    printf("1. Die Tür direkt vor dir.\n");
    printf("2. Die Tür zu deiner linken.\n");
    printf("3. Die Tür zu deiner rechten.\n");
    printf("4. Ich will doch ins Loch hüpfen...\n");
}

void room2scps()
{
    printf("Du entscheidest dich dazu, die Tür vor dir zu nähern und spürst mit jedem Schritt,\n");
    printf("den du dich der Tür näherst, dass sich vermutlich etwas besonderes dahinter verbirgt.\n\n");
    printf("Sowie du vor der Tür stehst, öffnest du ohne zu zögern die Tür und siehst in dem Raum\n");
    printf("4 weitere Türen die mit jeweils einer digitalen Kennzeichnung versehen sind:\n\n");
    printf("SCP-427\n Level 3 Keycard benötigt.\n");
    printf("SCP-860\n Level 3 Keycard benötigt.\n");
    printf("SCP-714\n Level 3 Keycard benötigt.\n");
    printf("SCP-1025\nLevel 3 Keycard benötigt.\n\n");
    printf("In der Mitte ist allerdings eine unbeschriftete Tür, die schon offen und unbeleuchtet ist.\n");
    printf("Was ist deine Entscheidung?\n\n");
    printf("Geh in den dunklen Raum\n");
    printf("Geh zurück zum Loch\n");
    printf("SCP-427 Kennzeichnung lesen\n");
    printf("SCP-860 Kennzeichnung lesen\n");
    printf("SCP-714 Kennzeichnung lesen\n");
    printf("SCP-1025 Kennzeichnung lesen\n\n");
    printf("(Gib dafür einen der folgenden Möglichkeiten ein, so wie die da stehen)\n");
}

void Main_Menu ()
{
    printf("Hauptmenü:\n");
    printf("1. Start\n");
    printf("2. Credits\n");
    printf("3. Ende\n");

    printf("\nWas möchten Sie gerne tun? \n");
    printf("Geben Sie dafür nur einer der Zahlen ein: \n");
}

void credits()
{
    printf("Dieses Spiel wurde von 3 Azubis der Unterstufe FiAe entwickelt \n\n");
    printf("Skript: Herr Meyer \n");
    printf("Spieltester: Herr Hübscher \n");
    printf("Coding: Herr Adams \n \n");
    printf("↓↓↓ Herr Meyer hat sich die Zeit genommen und uns seine Version der Website bereitgestellt! ↓↓↓\n");
    printf("Weblink Platzhalter \n\n");

    printf("Wir hoffen das ihnen das Spiel gefallen hat :-) \n");
}


lib/inventar.h
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

#define MAX_ITEMS 10
#define ITEM_NAME_LENGTH 30


// Define a structure for an item
typedef struct 
{
    char name[ITEM_NAME_LENGTH];
    int quantity;
} Item;

// Initialize the player's inventory
Item inventory[MAX_ITEMS];
int itemCount = 0;

/* 
    Function to add an item to the inventory
    Function arguments itemname for the name for the Item too add to the Inventory
    quantity for the position of the Item
*/
void addItem(const char* itemName, int quantity) {
    if (itemCount < MAX_ITEMS) {
        strcpy(inventory[itemCount].name, itemName);
        inventory[itemCount].quantity = quantity;
        itemCount++;
        printf("%s added to inventory.\n", itemName);
    } else {
        printf("Inventory is full. Cannot add %s.\n", itemName);
    }
}

// Function to remove an item from the inventory
void removeItem(const char* itemName) {
    int found = 0;
    for (int i = 0; i < itemCount; i++) {
        if (strcmp(inventory[i].name, itemName) == 0) {
            // Found the item, remove it by shifting elements
            for (int j = i; j < itemCount - 1; j++) {
                strcpy(inventory[j].name, inventory[j + 1].name);
                inventory[j].quantity = inventory[j + 1].quantity;
            }
            itemCount--;
            found = 1;
            printf("%s removed from inventory.\n", itemName);
            break;
        }
    }
    if (!found) {
        printf("Item not found in inventory: %s\n", itemName);
    }
}

// Function to display the player's inventory
void displayInventory() {
    printf("Inventory:\n");
    for (int i = 0; i < itemCount; i++) {
        printf("%s (%d)\n", inventory[i].name, inventory[i].quantity);
    }
}

lib/gameMechanics.c
#include <stdio.h>
#include <stdlib.h>
#include <stdbool.h>
#include "inventar.h"
#include "Story.c"


// Spielstart
void startGame() 
{
    printf("\n \nSpiel wird gestartet: \n \n");
    starting_room();
}

void SCP_FILEREADER(const char *dateiname)
{
    FILE *datei = fopen(dateiname, "r");

    if (datei)
    {
        char zeile[256];
        while (fgets(zeile, sizeof(zeile), datei))
        {
            printf("%s", zeile);
        }
    
        fclose(datei);
    }
    else
    {
        printf("\n\nDie Datei '%s' konnte nicht geöffnet werden, \n", dateiname);
    }
}


//Korrosives Loch

void easter_egg_larry()
{
    const char *larry = "Larry.txt";

    printf("\n\nDu springst elegant hinein und landest perfekt.\n");
    printf("Du schaust dich um und siehst SCP-106 (auch Larry genannt).\n\n");
    SCP_FILEREADER(larry);
    printf("\n\nLarry war nicht erfreut das du in sein Loch gesprungen bist und zögert nicht lange dich zu töten.\n\n");
    printf("Game Over (Du hast Larry gefunden! :D)\n\n");
    printf("Infos zu SCP-106 sind im folgenden Link → ");
    printf("file:///mnt/C/Users/adamsdav/Nextcloud/FIAE2023-1/Gruppenprojekte/SCP_Text-Based-Adventure_Game_Adams_Meyer_Huebscher/SCP-textadventure/SCP-Liste/SCP-106/SCP-106.html \n\n");
}

//Fehlermeldung sowie eine ungültige Eingabe getätigt wurde

void debug()
{
    const char *peanut = "peanut.txt";
    SCP_FILEREADER(peanut);
    printf("\n\nSCP-173 hat gesehen was du versucht hast und fand das nicht so toll...\n\n");
}


// Für Entscheidungssituationen

int path2_o (char choice)
{
    if (choice == 'j')
    {
        return true;
    }
    else
    if (choice == 'n')
    {
        return false;
    }
}

int path4_o (char choice)
{
    if(choice == 'l')
    {

    }
    else if (choice == 'v')
    {
        return true;
    }
    else if(choice == 'r')
    {

    }
    else if(choice == 'o')
    {
        return true;
    }
}

int path6_o (char choice)
{
    if(choice == 'v')
    {
        return true;
    }
    else if (choice == '1')
    {
        return true;
    }
    else if(choice == '2')
    {
        return true;
    }
    else if(choice == '3')
    {
        return true;
    }
    else if(choice == '4')
    {
        return true;
    }
    else if(choice == 'z')
    {
        return true;
    } 
}

//clear Funktion um das terminal aufzuräumen
void clear(){
    #if defined(__linux__) || defined(__unix__) || defined(__APPLE__)
        system("clear");
    #endif

    #if defined(_WIN32) || defined(_WIN64)
        system("cls");
    #endif
    }

game.c
#include <stdio.h>
#include <stdlib.h>
#include <stdbool.h>
#include "inventar.h"
#include "Story.c"
/*
Kürzel Siehe ReadMe Datei
*/

// SCP ASCII Funktion Definierung

void SCP(const char *dateiname)
{
    FILE *datei = fopen(dateiname, "r");

    if (datei)
    {
        char zeile[256];
        while (fgets(zeile, sizeof(zeile), datei))
        {
            printf("%s", zeile);
        }
    
        fclose(datei);
    }
    else
    {
        printf("\n\nDie Datei '%s' konnte nicht geöffnet werden, \n", dateiname);
    }
}

// Spielstart
void startGame() 
{
    printf("\n \nSpiel wird gestartet: \n \n");
    r1_s();
}

//Korrosives Loch

void easter_egg_larry()
{
    const char *dateiname2 = "Larry.txt";

    printf("\n\nDu springst elegant hinein und landest perfekt.\n");
    printf("Du schaust dich um und siehst SCP-106 (auch Larry genannt).\n\n");
    SCP(dateiname2);
    printf("\n\nLarry war nicht erfreut das du in sein Loch gesprungen bist und zögert nicht lange dich zu töten.\n\n");
    printf("Game Over (Du hast Larry gefunden! :D)\n\n");
    printf("Infos zu SCP-106 sind im folgenden Link → ");
    printf("file:///mnt/C/Users/adamsdav/Nextcloud/FIAE2023-1/Gruppenprojekte/SCP_Text-Based-Adventure_Game_Adams_Meyer_Huebscher/SCP-textadventure/SCP-Liste/SCP-106/SCP-106.html \n\n");
}

//Fehlermeldung sowie eine ungültige Eingabe getätigt wurde

void debug()
{
    const char *dateiname1 = "peanut.txt";
    SCP(dateiname1);
    printf("\n\nSCP-173 hat gesehen was du versucht hast und fand das nicht so toll...\n\n");
}

// Für Entscheidungssituationen

int path2_o (char choice)
{
    if (choice == 'j')
    {
        return true;
    }
    else
    if (choice == 'n')
    {
        return false;
    }
}

int path4_o (char choice)
{
    if(choice == 'l')
    {

    }
    else if (choice == 'v')
    {
        return true;
    }
    else if(choice == 'r')
    {

    }
    else if(choice == 'o')
    {
        return true;
    }
}

int path6_o (char choice)
{
    if(choice == 'v')
    {
        return true;
    }
    else if (choice == '1')
    {
        return true;
    }
    else if(choice == '2')
    {
        return true;
    }
    else if(choice == '3')
    {
        return true;
    }
    else if(choice == '4')
    {
        return true;
    }
    else if(choice == 'z')
    {
        return true;
    } 
}

//clear Funktion um das terminal aufzuräumen
void clear(){
    #if defined(__linux__) || defined(__unix__) || defined(__APPLE__)
        system("clear");
    #endif

    #if defined(_WIN32) || defined(_WIN64)
        system("cls");
    #endif
    }


int main()
{
    int choices;
    Main_Menu();
    scanf("%d",&choices);

switch (choices)
{
    case 1:    
        char* playerLocation[99];
        char choice;
        startGame();
        playerLocation[0] = "r1_s";
        scanf("%s",&choice);

    for(int i = 0; i != 99; i++)
    {
        //Raum h1_s
        if(strcmp(*playerLocation, "r1_s") == 0)
        {
            if(path2_o(choice))
            {
                h1_s();
                scanf("%s", &choice);
                playerLocation[0] = "h1_s";
            }
            else
            {
                r1_d_2();
                exit(1);
            }
        }
        
        //Raum h1_d_1
        if(strcmp(*playerLocation, "h1_s") == 0)
        {
            if(path2_o(choice))
            {
                h1_d_1();
                scanf("%s", &choice);
                playerLocation[0] = "h1_d_1";
            }
            else
            {
                h1_d_2();
                exit(2);
            }
        }

        //Raum h1_d_1 decision nach welchem Raum
        if(strcmp(*playerLocation, "h1_d_1") == 0)
        {
            if(path4_o(choice))
            {
                if(choice == 'l')
                {
                    scanf("%s", &choice);
                }
                else if(choice == 'v')
                {
                    h1_1_d_s();
                    scanf("%s", &choice);
                    playerLocation[0] = "h1_1_d_s";
                }
                else if(choice == 'r')
                {
                    scanf("%s", &choice);
                }
                else if(choice == 'o')
                {
                    easter_egg_larry();
                    exit(3);
                }
            }
        }

        //Raum h1_1_d_s decision zu welchem Raum
        if (strcmp(*playerLocation, "h1_1_d_s") == 0)
        {
            if(path6_o(choice))
            {
                if(choice == 'v')
                {
                    
                }
                else if (choice == '1')
                {
                    
                }
                else if(choice == '2')
                {
                    
                }
                else if(choice == '3')
                {
                    
                }
                else if(choice == '4')
                {
                    
                }
                else if(choice == 'z')
                {
                    h1_d_1();
                    scanf("%s", &choice);
                    playerLocation[0] = "h1_d_1";
                } 
            }
        }

    }





    
    break;

    case 2:
    credits();
    break;

    case 3:
    printf("Das Programm wird nun beendet \n");
    exit(0);
    break;

    default:
    debug();
    break;
return 0;
}


}